languages/dotnet/templates/RequestOptions.mustache

```diff
@@ -0,0 +1,73 @@
+{{! This file is picked from the v5.1.0 release of openapi-generator(https://github.com/OpenAPITools/openapi-generator/blob/v5.1.0/modules/openapi-generator/src/main/resources/csharp-netcore/RequestOptions.mustache) }}
+{{! Changes made: 1) Added ResponseReturnTypes map for different status code and return type combinations. }}
+{{>partial_header}}
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+
+namespace {{packageName}}.Client
+{
+    /// <summary>
+    /// A container for generalized request inputs. This type allows consumers to extend the request functionality
+    /// by abstracting away from the default (built-in) request framework (e.g. RestSharp).
+    /// </summary>
+    public class RequestOptions
+    {
+        /// <summary>
+        /// Parameters to be bound to path parts of the Request's URL
+        /// </summary>
+        public Dictionary<String, String> PathParameters { get; set; }
+
+        /// <summary>
+        /// Query parameters to be applied to the request.
+        /// Keys may have 1 or more values associated.
+        /// </summary>
+        public Multimap<String, String> QueryParameters { get; set; }
+
+        /// <summary>
+        /// Header parameters to be applied to to the request.
+        /// Keys may have 1 or more values associated.
+        /// </summary>
+        public Multimap<String, String> HeaderParameters { get; set; }
+
+        /// <summary>
+        /// Form parameters to be sent along with the request.
+        /// </summary>
+        public Dictionary<String, String> FormParameters { get; set; }
+
+        /// <summary>
+        /// File parameters to be sent along with the request.
+        /// </summary>
+        public Dictionary<String, Stream> FileParameters { get; set; }
+
+        /// <summary>
+        /// Cookies to be sent along with the request.
+        /// </summary>
+        public List<Cookie> Cookies { get; set; }
+
+        /// <summary>
+        /// Response return types for each status code.
+        /// </summary>
+        public Dictionary<int, Type> ResponseReturnTypes { get; set; }
```

Comment:
This file was copied from OpenAPI project and only the ResponseReturnTypes property was added.

Reply:
Can you move this file to init branch so that this PR would only show the addition of the `ResponseReturnTypes` property